lib/livebook_web/live/hub/teams/deployment_group_live.ex

```diff
@@ -179,6 +215,39 @@ defmodule LivebookWeb.Hub.Teams.DeploymentGroupLive do
      |> update_dockerfile()}
   end
 
+  def handle_event("add_agent_key", _, socket) do
+    on_confirm = fn socket ->
+      hub = Livebook.Hubs.fetch_hub!(socket.assigns.hub.id)
+      deployment_group = socket.assigns.deployment_group
+
+      case Teams.create_agent_key(hub, deployment_group) do
+        :ok ->
+          socket
+          |> put_flash(:success, "Agent key created successfully")
+          |> push_patch(to: ~p"/hub/#{hub.id}/deployment-groups/edit/#{deployment_group.id}")
+
+        {:error, _changeset} ->
+          put_flash(
+            socket,
+            :error,
+            "Something went wrong, try again later or please file a bug if it persists"
+          )
+
+        {:transport_error, reason} ->
+          put_flash(socket, :error, reason)
+      end
+    end
+
+    {:noreply,
+     confirm(socket, on_confirm,
+       title: "Create agent key",
+       description: "Are you sure you want to create a new agent key?",
```

Comment:
Nit: "are you sure?" makes the user second-guess the action, for a non-destructive action we can just expand on the end result.

```suggestion
       description: "This will create a new agent key for this deployment group.",
```